templates/identity/email-password/forgot-password-response.html.ejs

```diff
@@ -1,13 +1,20 @@
 <h1>Email sent</h1>
-<form method="post">
-  <p>If your account exists, an email has been sent with a link to reset your password.</p>
-  <p>If you do not receive your email in a couple of minutes, check your spam folder or click the link below to send another email.</p>
+<p>If your account exists, an email has been sent with a link to reset your password.</p>
+<p>If you do not receive your email in a couple of minutes, check your spam folder or try sending another email.</p>
 
-  <input type="hidden" name="email" value="<%= email %>" />
+<ul class="actions">
+  <li><a id="login-link" href="" class="link">Back to Log In</a></li>
+  <li><a id="forgot-link"  href="" class="link">Back to Forgot Password</a></li>
+</ul>
 
-  <p class="actions"><a href="<%= controls.login %>">Back to Log In</a></p>
-
-  <p class="actions">
-    <button type="submit" name="submit" class="link">Send Another Email</button>
-  </p>
-</form>
+<script>
+  // Get controls
+  (async() => {
+    const res = await fetch('<%= idpIndex %>');
```

Comment:
Code injection warning; needs to be (in order) 1) JavaScript-escaped 2) HTML-escaped.

Reply:
`idpIndex` is a hardcoded value in the server (being the entrypoint of the API). How could injection occur here? Also if it's escaped that URL would not be usable anymore here.

Reply:
> Also if it's escaped that URL would not be usable anymore here.

It would; any string inserted in JavaScript code needs to be JS-escaped; JS will de-escape upon execution.

> How could injection occur here?

Config; not a big attack vector, rather sensible practice to escape.

Probs are URLs containing `'` and `</script>`

Reply:
HTML escaping already happens by using `<%=` so the only potential risk is if the injected value breaks out of the string and adds some extra JS there. Not sure if ejs already escape quotes, will need to check. If not we could potentially prevent it by something like `const res = await fetch('<%= idpIndex.replace(/'/g, '') %>');`.

Reply:
Did some tests. First of all, if someone did change the config this would mostly break the entire IDP component of the server since this same URL is used as the base for all IDP paths. E.g., the login page is relative to that URL so if you somehow changed the index to `http://localhost:3000/idp/'</script><script>hack hack hack</script>`, your login page would now be located at `http://localhost:3000/idp/'</script><script>hack hack hack</script>/login`. Although I guess you could also rewrite that entire part of the config if you're injecting code anyway but that's starting to become a lot of work :smile:.

But even then, I checked and this is the output ejs generates, so it also escapes the quotes when doing HTML escaping, so it was already fine.
```javascript
let res = await fetch('http://localhost:3000/idp/&#39;&lt;/scrip&gt;&lt;script&gt;hackhackhack&lt;/script&gt;', { headers: {accept: 'application/json' }});
```